README.md
# development

TODO: Enter the cookbook description here.



CHANGELOG.md
# development CHANGELOG

This file is used to list changes made in each version of the development cookbook.

# 0.1.0

Initial release.

- change 0
- change 1



spec/spec_helper.rb
require 'chefspec'
require 'chefspec/policyfile'
at_exit{ChefSpec::Coverage.report!}


spec/unit/recipes/default_spec.rb
#
# Cookbook:: development
# Spec:: default
#
# Copyright:: 2019, The Authors, All Rights Reserved.

require 'spec_helper'

describe 'development::default' do
  context 'When all attributes are default, on Ubuntu/xenial64' do
  # for a complete list of available platforms and versions see:
  # https://github.com/chefspec/fauxhai/blob/master/PLATFORMS.md
  platform 'ubuntu', '18.04'
  end
    it 'converges successfully' do
      expect { chef_run }.to_not raise_error
    end
    it 'should update the sources list' do
      expect(chef_run).to update_apt_update 'update_sources'
    end
    it 'should install python' do
      expect(chef_run).to install_package 'python'
    end
    it 'should install pip for python' do
      expect(chef_run).to install_package 'python-pip'
    end
    it 'should install two plugins for gnureadline' do
      expect(chef_run).to install_package 'libncurses5-dev'
      expect(chef_run).to install_package 'libffi-dev'
    end
    it 'should install flask version 0.10.1' do
      expect(chef_run).to run_execute('python-pip install plugin flask')
    end
    it 'should install flask Jinja2 version 2.7.3' do
      expect(chef_run).to run_execute('python-pip install plugin Jinja2')
    end
    it 'should install markupsafe version 0.23' do
      expect(chef_run).to run_execute('python-pip install plugin markupsafe')
    end
    it 'should install werkzeug version 0.9.6' do
      expect(chef_run).to run_execute('python-pip install plugin werkzeug')
    end
    it 'should install gnureadline version 6.3.3' do
      expect(chef_run).to run_execute('python-pip install plugin gnureadline')
    end
    it 'should install itsdangerous version 0.24' do
      expect(chef_run).to run_execute('python-pip install plugin itsdangerous')
    end
    it 'should install rauth version 0.7.0' do
      expect(chef_run).to run_execute('python-pip install plugin rauth')
    end
    it 'should install requests version 2.3.0' do
      expect(chef_run).to run_execute('python-pip install plugin requests')
    end
    it 'should install wsgiref version 0.1.2' do
      expect(chef_run).to run_execute('python-pip install plugin wsgiref')
    end
    it 'should install gunicorn version 18.0' do
      expect(chef_run).to run_execute('python-pip install plugin gunicorn')
    end
    it 'should install Flask-SSLify version 0.1.4' do
      expect(chef_run).to run_execute('python-pip install plugin Flask-SSLify')
    end
    it 'should install pytest version 2.5.2' do
      expect(chef_run).to run_execute('python-pip install test plugin pytest')
    end
    it 'should install pytest-cov version 1.6' do
      expect(chef_run).to run_execute('python-pip install test plugin pytest-cov')
    end
    it 'should install betamax version 0.4.0' do
      expect(chef_run).to run_execute('python-pip install test plugin betamax')
    end
    it 'should install flake8 version 2.1.0' do
      expect(chef_run).to run_execute('python-pip install test plugin flake8')
    end
    it 'should install pep8 version 1.5.6' do
      expect(chef_run).to run_execute('python-pip install test plugin pep8')
    end
    it 'should install pyflakes version 0.8.1' do
      expect(chef_run).to run_execute('python-pip install test plugin pyflakes')
    end
    it 'should install coveralls version 0.4.2' do
      expect(chef_run).to run_execute('python-pip install test plugin coveralls')
    end
end


test/integration/default/default_test.rb
# InSpec test for recipe development::default

# The InSpec reference, with examples and extensive documentation, can be
# found at https://www.inspec.io/docs/reference/resources/

describe package('python') do
  it { should be_installed }
end

describe package('python-pip') do
  it { should be_installed }
end

describe package('libncurses5-dev') do
  it {should be_installed}
end

describe package('libffi-dev') do
  it {should be_installed}
end

describe pip('flask') do
  it { should be_installed }
  its('version') { should eq '0.10.1' }
end

describe pip('Jinja2') do
  it { should be_installed }
  its('version') { should eq '2.7.3' }
end

describe pip('markupsafe') do
  it { should be_installed }
  its('version') { should eq '0.23' }
end

describe pip('gnureadline') do
  it { should be_installed }
  its('version') { should eq '6.3.3' }
end

describe pip('itsdangerous') do
  it { should be_installed }
  its('version') { should eq '0.24' }
end

describe pip('rauth') do
  it { should be_installed }
  its('version') { should eq '0.7.0' }
end

describe pip('requests') do
  it { should be_installed }
  its('version') { should eq '2.3.0' }
end

describe pip('wsgiref') do
  it { should be_installed }
  its('version') { should eq '0.1.2' }
end

describe pip('gunicorn') do
  it { should be_installed }
  its('version') { should eq '18.0' }
end

describe pip('Flask-SSLify') do
  it { should be_installed }
  its('version') { should eq '0.1.4' }
end

describe pip('pytest') do
  it { should be_installed }
  its('version') { should eq '2.5.2' }
end

describe pip('pytest-cov') do
  it { should be_installed }
  its('version') { should eq '1.6' }
end

describe pip('betamax') do
  it { should be_installed }
  its('version') { should eq '0.4.0' }
end

describe pip('flake8') do
  it { should be_installed }
  its('version') { should eq '2.1.0' }
end

describe pip('pep8') do
  it { should be_installed }
  its('version') { should eq '1.5.6' }
end

describe pip('pyflakes') do
  it { should be_installed }
  its('version') { should eq '0.8.1' }
end

describe pip('coveralls') do
  it { should be_installed }
  its('version') { should eq '0.4.2' }
end


recipes/default.rb
#
# Cookbook:: development
# Recipe:: default
#
# Copyright:: 2019, The Authors, All Rights Reserved.

apt_update 'update_sources' do
  action :update
end

package 'python' do
  action :install
end

package 'python-pip' do
  action :install
end

package 'libncurses5-dev' do
  action :install
end

package 'libffi-dev' do
  action :install
end

execute 'python-pip install plugin flask' do
  command 'pip install flask==0.10.1'
end
execute 'python-pip install plugin Jinja2' do
  command 'pip install Jinja2==2.7.3'
end
execute 'python-pip install plugin markupsafe' do
  command 'pip install markupsafe==0.23'
end
execute 'python-pip install plugin werkzeug' do
  command 'pip install werkzeug==0.9.6'
end
execute 'python-pip install plugin gnureadline' do
  command 'pip install gnureadline==6.3.3'
end
execute 'python-pip install plugin itsdangerous' do
  command 'pip install itsdangerous==0.24'
end
execute 'python-pip install plugin rauth' do
  command 'pip install rauth==0.7.0'
end
execute 'python-pip install plugin requests' do
  command 'pip install requests==2.3.0'
end
execute 'python-pip install plugin wsgiref' do
  command 'pip install wsgiref==0.1.2'
end
execute 'python-pip install plugin gunicorn' do
  command 'pip install gunicorn==18.0'
end
execute 'python-pip install plugin Flask-SSLify' do
  command 'pip install Flask-SSLify==0.1.4'
end
execute 'python-pip install test plugin pytest' do
  command 'pip install pytest==2.5.2'
end
execute 'python-pip install test plugin pytest-cov' do
  command 'pip install pytest-cov==1.6'
end
execute 'python-pip install test plugin betamax' do
  command 'pip install betamax==0.4.0'
end
execute 'python-pip install test plugin flake8' do
  command 'pip install flake8==2.1.0'
end
execute 'python-pip install test plugin pep8' do
  command 'pip install pep8==1.5.6'
end
execute 'python-pip install test plugin pyflakes' do
  command 'pip install pyflakes==0.8.1'
end
execute 'python-pip install test plugin coveralls' do
  command 'pip install coveralls==0.4.2'
end
